src/services/Xml.php

```diff
@@ -443,11 +443,16 @@ public function customOrderFields(\SimpleXMLElement $order_xml, Order $order)
             OrderFieldEvent::FIELD_CUSTOM_FIELD_1,
             OrderFieldEvent::FIELD_CUSTOM_FIELD_2,
             OrderFieldEvent::FIELD_CUSTOM_FIELD_3,
-            OrderFieldEvent::FIELD_INTERNAL_NOTES,
-            OrderFieldEvent::FIELD_CUSTOMER_NOTES,
             OrderFieldEvent::FIELD_GIFT,
+        ];
+        
+        $longCustomFields = [
             OrderFieldEvent::FIELD_GIFT_MESSAGE,
+            OrderFieldEvent::FIELD_INTERNAL_NOTES,
+            OrderFieldEvent::FIELD_CUSTOMER_NOTES,
         ];
+        
+        $customFields = array_merge($customFields, $longCustomFields);
```

Comment:
Instead of using a conditional to set the field limit, we could specify the limit per field in the array:

```php
$customFields = [
  [OrderFieldEvent::FIELD_CUSTOM_FIELD_1, 100],
  [OrderFieldEvent::FIELD_GIFT_MESSAGE, 1000],
  // etc
];

...

foreach ($customFields as list($fieldName, $charLimit)) {
  // ...
}
```

If the spec changed in the future we can change it for just fields which are affected.

Reply:
I like it, nice and clean! I've update my pull request.